src/helm/benchmark/model_metadata_registry.py

```diff
@@ -55,17 +135,77 @@ def register_model_metadata_from_path(path: str) -> None:
-        model = Model(
-            group="none",  # TODO: Group should be part of model deployment, not model
-            name=model_metadata.name,
-            tags=model_metadata.tags,
-        )
-        MODEL_NAME_TO_MODEL[model_metadata.name] = model
-        ALL_MODELS.append(model)
+        register_model_metadata(model_metadata)
+
+
+def register_model_metadata(model_metadata: ModelMetadata) -> None:
+    """Register a single model configuration."""
+    # hlog(f"Registered model metadata {model_metadata.name}")
+    ALL_MODELS_METADATA.append(model_metadata)
+    MODEL_NAME_TO_MODEL_METADATA[model_metadata.name] = model_metadata
 
 
 def maybe_register_model_metadata_from_base_path(base_path: str) -> None:
     """Register model metadata from prod_env/model_metadata.yaml"""
     path = os.path.join(base_path, MODEL_METADATA_FILE)
     if os.path.exists(path):
         register_model_metadata_from_path(path)
+
+
+# ===================== UTIL FUNCTIONS ==================== #
+def get_model_metadata(model_name: str) -> ModelMetadata:
+    """Get the `Model` given the name."""
+    register_metadatas_if_not_already_registered()
+    if model_name not in MODEL_NAME_TO_MODEL_METADATA:
+        raise ValueError(f"No model with name: {model_name}")
+
+    return MODEL_NAME_TO_MODEL_METADATA[model_name]
+
+
+def get_model_creator_organization(model_name: str) -> str:
+    """Get the model's group given the name."""
+    model: ModelMetadata = get_model_metadata(model_name)
+    return model.creator_organization
+
+
+def get_all_models() -> List[str]:
+    """Get all model names."""
+    register_metadatas_if_not_already_registered()
+    return list(MODEL_NAME_TO_MODEL_METADATA.keys())
+
+
+def get_models_by_creator_organization(organization: str) -> List[str]:
+    """
+    Gets models by creator organization.
+    Example:   ai21   =>   ai21/j1-jumbo, ai21/j1-grande, ai21-large.
+    """
+    register_metadatas_if_not_already_registered()
+    return [model.name for model in ALL_MODELS_METADATA if model.creator_organization == organization]
+
+
+def get_model_names_with_tag(tag: str) -> List[str]:
+    """Get all the name of the models with tag `tag`."""
+    register_metadatas_if_not_already_registered()
+    return [model.name for model in ALL_MODELS_METADATA if tag in model.tags]
+
+
+def get_all_text_models() -> List[str]:
+    """Get all text model names."""
+    return get_model_names_with_tag(TEXT_MODEL_TAG)
+
+
+def get_all_code_models() -> List[str]:
+    """Get all code model names."""
+    return get_model_names_with_tag(CODE_MODEL_TAG)
+
+
+def get_all_instruction_following_models() -> List[str]:
+    """Get all instruction-following model names."""
+    return get_model_names_with_tag(INSTRUCTION_FOLLOWING_MODEL_TAG)
+
+
+def register_metadatas_if_not_already_registered() -> None:
+    global METADATAS_REGISTERED
+    if not METADATAS_REGISTERED:
+        maybe_register_model_metadata_from_base_path(CONFIG_PATH)
```

Comment:
This doesn't work with pip-installed HELM because it will attempt to look for `src/helm/config` relative to the _current working directory_, not the _package installation directory_.

You need to do something like this `importlib_resources` [example](https://github.com/stanford-crfm/helm/blob/3648c7f25968a32a4c917e3e6bfb4001f34b6042/src/helm/benchmark/presentation/contamination.py#L82) instead.